lib/shoryuken/fetcher.rb

```diff
@@ -5,23 +5,21 @@ class Fetcher
     FETCH_LIMIT = 10
 
     def fetch(queue, available_processors)
-      watchdog('Fetcher#fetch died') do
-        started_at = Time.now
-
-        logger.debug { "Looking for new messages in '#{queue}'" }
-
-        begin
-          limit = available_processors > FETCH_LIMIT ? FETCH_LIMIT : available_processors
-
-          sqs_msgs = Array(receive_messages(queue, limit))
-          logger.info { "Found #{sqs_msgs.size} messages for '#{queue.name}'" }
-          logger.debug { "Fetcher for '#{queue}' completed in #{elapsed(started_at)} ms" }
-          sqs_msgs
-        rescue => ex
-          logger.error { "Error fetching message: #{ex}" }
-          logger.error { ex.backtrace.first }
-          []
-        end
+      started_at = Time.now
+
+      logger.debug { "Looking for new messages in '#{queue}'" }
+
+      begin
+        limit = available_processors > FETCH_LIMIT ? FETCH_LIMIT : available_processors
+
+        sqs_msgs = Array(receive_messages(queue, limit))
+        logger.info { "Found #{sqs_msgs.size} messages for '#{queue.name}'" }
```

Comment:
@tjsingleton I haven't, but it seems to be a good idea. The only one issue is because the process will may look like isn't running/moving without that message, in case all queues are empty. WDYT?